engine/schema/src/main/java/com/cloud/upgrade/DatabaseUpgradeChecker.java

```diff
@@ -208,6 +208,7 @@ public DatabaseUpgradeChecker() {
                 .next("4.16.1.0", new Upgrade41610to41700())
                 .next("4.16.1.1", new Upgrade41610to41700())
                 .next("4.17.0.0", new Upgrade41700to41710())
+                .next("4.17.0.1", new Upgrade41700to41710())
```

Comment:
Not an issue, if somehow it won't matter or doesn't matter; and this somehow runs while upgrading from 4.17.0.1.

Reply:
it should @rohityadavcloud , but that is as designed. db upgrades historically ignore the last digit. Did you add this PR because of an upgrade problem @rohityadavcloud ?

Reply:
@rohityadavcloud this is needed. Upgrading from 4.17.0.1 to 4.17.1.0 doesn't apply schema changes
```
2022-09-23 09:17:30,348 INFO  [o.a.c.s.l.CloudStackExtendedLifeCycle] (main:null) (logid:) Running system integrity checker com.cloud.upgrade.DatabaseUpgradeChecker@e29f4f6
2022-09-23 09:17:30,348 INFO  [c.c.u.DatabaseUpgradeChecker] (main:null) (logid:) Grabbing lock to check for database upgrade.
2022-09-23 09:17:30,368 DEBUG [c.c.u.d.VersionDaoImpl] (main:null) (logid:) Checking to see if the database is at a version before it was the version table is created
2022-09-23 09:17:30,381 DEBUG [c.c.u.SystemVmTemplateRegistration] (main:null) (logid:) Looking for file [ ./engine/schema/dist/systemvm-templates/metadata.ini ] in the classpath.
2022-09-23 09:17:30,381 DEBUG [c.c.u.SystemVmTemplateRegistration] (main:null) (logid:) Looking for file [ /usr/share/cloudstack-management/templates/systemvm/metadata.ini ] in the classpath.
2022-09-23 09:17:30,489 INFO  [c.c.u.DatabaseUpgradeChecker] (main:null) (logid:) DB version = 4.17.0.1 Code Version = 4.17.1.0
2022-09-23 09:17:30,490 INFO  [c.c.u.DatabaseUpgradeChecker] (main:null) (logid:) Database upgrade must be performed from 4.17.0.1 to 4.17.1.0
2022-09-23 09:17:30,520 DEBUG [c.c.u.DatabaseUpgradeChecker] (main:null) (logid:) Running upgrade NoopDbUpgrade to upgrade from 4.17.0.1-4.17.1.0 to 4.17.1.0
2022-09-23 09:17:30,880 INFO  [c.c.u.DatabaseUpgradeChecker] (main:null) (logid:) Cleanup upgrade NoopDbUpgrade to upgrade from 4.17.0.1-4.17.1.0 to 4.17.1.0
2022-09-23 09:17:30,900 DEBUG [c.c.u.DatabaseUpgradeChecker] (main:null) (logid:) Upgrade completed for version 4.17.1.0
2022-09-23 09:17:30,901 INFO  [o.a.c.s.l.CloudStackExtendedLifeCycle] (main:null) (logid:) Configuring CloudStack Components
2022-09-23 09:17:30,903 INFO  [o.a.c.s.l.CloudStackExtendedLifeCycle] (main:null) (logid:) Done Configuring CloudStack Components
2022-09-23 09:17:30,903 INFO  [o.a.c.s.m.m.i.DefaultModuleDefinitionSet] (main:null) (logid:) Loaded module context [system] in 1653 ms
```